commons/zenoh-buffers/src/zslicemut.rs

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use core::ops::{Deref, DerefMut, Range};
+
+/*************************************/
+/*             ZSLICEMUT             */
+/*************************************/
+pub struct ZSliceMut<'a> {
+    slice: &'a mut [u8],
+    start: &'a mut usize,
+    end: &'a mut usize,
+}
+
+impl<'a> ZSliceMut<'a> {
+    pub(crate) fn new(slice: &'a mut [u8], start: &'a mut usize, end: &'a mut usize) -> Self {
+        Self { slice, start, end }
+    }
+
+    pub fn range(&self) -> Range<usize> {
+        *self.start..*self.end
+    }
+
+    pub fn set_range(&mut self, range: Range<usize>) -> bool {
+        match self.slice.len() >= range.end {
+            true => {
+                *self.start = range.start;
+                *self.end = range.end;
+                true
+            }
+            false => false,
+        }
+    }
+
+    pub fn available_len(&self) -> usize {
+        self.slice.len()
+    }
+
+    pub fn len(&self) -> usize {
+        *self.end - *self.start
+    }
+
+    #[inline]
+    #[must_use]
+    pub const fn is_empty(&self) -> bool {
+        *self.end == *self.start
+    }
+}
+
+impl<'a> Drop for ZSliceMut<'a> {
```

Comment:
What's the purpose of implementing an empty `Drop`?

Reply:
I just forgot the todo here. There is an additional write-locking mechanism intended (to prevent from Policy-triggered deallocation), but I decided to add it at optimization stage, as it correlates with som header structure changes I planned. The release for this lock is intended in this Drop